iroh-net/src/magicsock/endpoint.rs

```diff
@@ -1112,6 +1193,15 @@ impl PeerMap {
         self.by_ip_port.insert(*ipp, id);
     }
 
+    /// Saves the known peer info to the given path, returning the number of peers persisted.
+    pub(super) fn save_to_file(&self, path: &Path) -> anyhow::Result<usize> {
+        let known_peers = self.known_peers();
```

Comment:
This allocates the whole thing in memory, effectively duplicating memory usage for this moment. I wonder if we could serialize it in an iterator/streaming fashion. eg sth like this

```rust
let mut file = std::fs::File::open(path)?;
file.write_all(HEADER)?;
let mut buffer = Vec::new();
for peer in &self.known_peers() {
  let ser = postcard::to_slice(peer, &mut buffer)?;
  file.write_all(ser)?;
}
file.flush()?;
```

Reply:
(and make `self.known_peers()` return an iterator instead of a vec)

Reply:
didn't get to do this but will look into it